test/integration/toolchains/cmake/cmakedeps2/test_cmakedeps.py

```diff
@@ -211,6 +215,134 @@ class TestConan(ConanFile):
     assert "WARN: Using the new CMakeConfigDeps generator" in c.out
 
 
+@pytest.mark.skipif(platform.system() != "Linux", reason="Only linux")
+def test_cpp_info_sources():
+    c = TestClient()
+    c.run("new cmake_lib -d name=hello -d version=1.0")
+
+    cmakelists = textwrap.dedent("""
+        cmake_minimum_required(VERSION 3.15)
+        project(hello CXX)
+
+        add_library(hello INTERFACE)
+        target_include_directories(hello INTERFACE include)
+        set_target_properties(hello PROPERTIES PUBLIC_HEADER "include/hello.h")
+        target_sources(hello INTERFACE
+                       "include/hello.h"
+                       "src/hello.cpp")
```

Comment:
Probably this can be simplified, not really necessary.
The package recipe can be much simpler, even without CMake code, as if it was a header-only library.
We are not interested in this, just on the ``package_info()`` definition of sources and consumption